elisp/intero.el

```diff
@@ -1661,6 +1699,9 @@ Automatically performs initial actions in SOURCE-BUFFER, if specified."
       (set-process-query-on-exit-flag process nil)
       (process-send-string process ":set -fobject-code\n")
       (process-send-string process ":set prompt \"\\4\"\n")
+      ;; The following settings give us expanded splices on load
+      (process-send-string process ":set -XTemplateHaskell\n")
+      (process-send-string process ":set -ddump-splices\n")
```

Comment:
I think we should move this into `intero-expand-splice-at-point`, using a `intero-blocking-call`. I think that dump-splices significantly increases output size and in proportion reduces performance of flycheck reporting.

Because `:set -ddump-splices` can be run repeatedly to no ill effects, we should "lazily" enable it as needed. That way, if someone's doing no template-haskell development, then they don't pay the cost. If someone is working on template-haskell and they want to see the splice, they can pay for it.

Unfortunately there seems no way to turn off flags, that I can see in GHC's documentation, so once you turn it on, you can't turn it off.